Hide unnecessarily exported symbols

Chibi libraries (accidentally) make some internal symbols public.
We don't want this so mark them with "static" to hide them.

"gc_heap_err_str" and "load_image_header" are just brain farts.

"_huff_tab*" symbols are trickier in that they are used in both
sexp.c (locally) and in SRFI 95 (qsort.c), which expect them
to be exported when built as static library. That's why we only
mark them with __attribute__((visibility("hidden"))) so that
they are hidden from shared library exports, but are visible
when linking static libraries.

All public symbols are prefixed with "sexp_*".

--- a/gc_heap.c
+++ b/gc_heap.c
@@ -7,7 +7,7 @@
 #if SEXP_USE_IMAGE_LOADING
 
 #define ERR_STR_SIZE 256
-char gc_heap_err_str[ERR_STR_SIZE];
+static char gc_heap_err_str[ERR_STR_SIZE];
 
 
 static sexp_uint_t sexp_gc_allocated_bytes (sexp ctx, sexp *types, size_t types_cnt, sexp x) {
@@ -573,7 +573,7 @@
 }
 
 
-int load_image_header(FILE *fp, struct sexp_image_header_t* header) {
+static int load_image_header(FILE *fp, struct sexp_image_header_t* header) {
   if (!fp || !header) { return 0; }
   
   if (fread(header, sizeof(struct sexp_image_header_t), 1, fp) != 1) {
--- a/include/chibi/sexp-hufftabs.c
+++ b/include/chibi/sexp-hufftabs.c
@@ -1,5 +1,7 @@
 /* auto-generated by ./symhuff.scm on Sun Feb 25 23:12:52 2007 */
 
+#pragma GCC visibility push(hidden)
+
 char _huff_tab21[] = {
   '\x01',  '\x00',  '\x03',  '\x00',  '\x02',  '\x00',  '\x04',  '\x00',
 };
@@ -90,3 +92,4 @@
      'e',     't',
 };
 
+#pragma GCC visibility pop
--- a/include/chibi/sexp-hufftabdefs.h
+++ b/include/chibi/sexp-hufftabdefs.h
@@ -1,3 +1,4 @@
+#pragma GCC visibility push(hidden)
 
 char _huff_tab1[8], _huff_tab2[8], _huff_tab3[2], _huff_tab4[2],
   _huff_tab5[4], _huff_tab6[2], _huff_tab7[4], _huff_tab8[4],
@@ -5,3 +6,5 @@
   _huff_tab13[8], _huff_tab14[2], _huff_tab15[8], _huff_tab16[8],
   _huff_tab17[8], _huff_tab18[8], _huff_tab19[4], _huff_tab20[8],
   _huff_tab21[8];
+
+#pragma GCC visibility pop
--- a/include/chibi/sexp-hufftabs.h
+++ b/include/chibi/sexp-hufftabs.h
@@ -1,5 +1,7 @@
 /* auto-generated by ./symhuff.scm on Sun Feb 25 23:12:52 2007 */
 
+#pragma GCC visibility push(hidden)
+
 char _huff_tab21[] = {
   '\x01',  '\x00',  '\x03',  '\x00',  '\x02',  '\x00',  '\x04',  '\x00',
 };
@@ -90,3 +92,4 @@
      'e',     't',
 };
 
+#pragma GCC visibility pop
